### PR TITLE
Upgrade python to 3.10

### PR DESCRIPTION
**Pull request by Yuyuutsu**
### What is the context of this PR?
This PR is upgrading the python version from 3.9 to 3.10 to ensure we have the latest features. Dockerfile and pip file are also updated.  
Black dev-package is also upgraded to 22.8.0

### How to review 
Check if tests and Makefile scripts are working fine.

### CODE REVIEW

**Comment by Yuyuutsu** [comment]
[Upgrade non-runner Python repos to 3.10 (2 SP)](https://trello.com/c/B9pw2uPW/5229-upgrade-non-runner-python-repos-to-310-2-sp)

**Comment by Yuyuutsu** [comment]
[Upgrade non-runner Python repos to 3.10 (2 SP)](https://trello.com/c/B9pw2uPW/5229-upgrade-non-runner-python-repos-to-310-2-sp)

**Review by petechd — APPROVED** [review]
Tested this in daily-test pipeline (and locally).

File: scripts/visualise_results.py [hunk 1]
```diff
@@ -9,7 +9,7 @@
 
 
 def plot_data(df, number_of_days_to_plot):
-    plt.style.use('seaborn-poster')
+    plt.style.use('fast')
```
Comment (petechd) [inline scripts/visualise_results.py:12]: @Yuyuutsu changed it so we don't get depreciation warning.

File: Pipfile [hunk 2]
```diff
@@ -4,7 +4,7 @@ verify_ssl = true
 name = "pypi"
 
 [dev-packages]
-black = "==19.10b0"
+black = "==22.8.0"
```
Comment (MebinAbraham) [inline Pipfile:7]: I suggest no longer pinning, we shoud use latest.

Comment (Yuyuutsu) [reply Pipfile]: No longer pinned 👍

**Review by berroar — APPROVED** [review]

**Pull request merged**